launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -369,8 +371,20 @@ void FirstLaunchView::extractGogData()
 
 		QString tmpFileExe = tempDir.filePath("h3_gog.exe");
 		QString tmpFileBin = tempDir.filePath("h3_gog-1.bin");
+#ifdef VCMI_ANDROID
+		auto copyFunc = [](std::string src, std::string dst)
+		{
+			CAndroidVMHelper vmHelper;
+			vmHelper.callCustomMethod(CAndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "copyFileFromUri", "(Ljava/lang/String;Ljava/lang/String;)V", [src, dst](JNIEnv * env, jclass javaHelper, jmethodID methodId){
+				env->CallStaticVoidMethod(javaHelper, methodId, env->NewStringUTF(src.c_str()), env->NewStringUTF(dst.c_str()));
+			}, true);
```

Comment:
not sure why you decided to write something out of your head instead of checking how it's already done in the launcher, for example:
https://github.com/vcmi/vcmi/blob/6e032946337ccaaeb189cc5f01d64f8299068305/launcher/prepare.cpp#L63

and it doesn't involve any lambdas, just pass proper objects

Reply:
To be honest. I wasn't aware of this function. I thought all functions that are called by native c++ code would be in NativeFunctions.

Feel free to provide tested code to use suggested approach.

I don't have any android build environment on my pc. The build instructions for android never worked for me. And without working ccache it's a pain to use CI for this.
Don't want to spend another hours on this. Was already too time consuming. But we needed a solution for this.

Reply:
> I don't have any android build environment on my pc. The build instructions for android never worked for me

have you tried asking for help / guidance?

> And without working ccache it's a pain to use CI for this.

are you in a hurry? We're not releasing another update in a few hours :) I suggest to configure local environment to speed things up.

Reply:
> are you in a hurry?

Not really.
But prioritys are shifting. Winter holiday season is over. So not sure how much time left in the next weeks besides small fixes. And would be great to fix this in the next 1.6.x as we have many reports on this problem.